AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -1111,6 +1129,28 @@ - (BOOL)openURL:(NSURL *)url {
   return YES;
 }
 
+- (void)setUpdateTrack:(MSUpdateTrack)updateTrack {
+  @synchronized(self) {
+    if (![MSDistributeUtil isValidUpdateTrack:updateTrack]) {
+      MSLogError([MSDistribute logTag], @"Invalid argument passed to updateTrack.");
+      return;
+    }
+    if (_updateTrack != updateTrack) {
```

Comment:
You can start the synchronized here after the validation maybe. I think the validation doesn't need to be synchronized.

Reply:
OK.